2.4 blog post (#11457)
prettier/prettier: changelog_unreleased/blog-post-intro.md

---

> author: "Sosuke Suzuki"

Reviewer: ```suggestion
author: "Sosuke Suzuki (@sosukesuzuki)"
```

authorURL: "https://github.com/sosukesuzuki"
title: "Prettier 2.4: new bracketSameLine option and TypeScript 4.4 support!"
---

> This release adds new `bracketSameLine` option! And `jsxBracketSameLine` is deprecated.`bracketSameLine` is generic version of the `jsxBracketSameLine` option that works for HTML, JSX, Vue, and Angular.

Reviewer: Not 100% happy with “which” in this suggestion but I can’t think of anything better at the moment.

```suggestion
This release renames the `jsxBracketSameLine` option to `bracketSameLine`, which supports HTML, Vue, and Angular in addition to JSX. The old name has been deprecated.
```

> Also we supports [TypeScript 4.4](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-4.html). This allows for the formatting of Stage 4 [Class Static Blocks](https://github.com/tc39/proposal-class-static-block).

Reviewer: ```suggestion
We’ve also added support for [TypeScript 4.4](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-4.html), including new syntax features such as [class `static` blocks](https://github.com/tc39/proposal-class-static-block).
```

> If you enjoy Prettier and would like to support our work, consider supporting us directly via [our OpenCollective](https://opencollective.com/prettier) or by supporting the projects we depend on, including [typescript-eslint](https://opencollective.com/typescript-eslint), [remark](https://opencollective.com/unified), and [Babel](https://opencollective.com/babel).

Reviewer: After looking at this again the repetition of “support” doesn’t sound right, oops!

```suggestion
If you enjoy Prettier and would like to support our work, consider sponsoring us directly via [our OpenCollective](https://opencollective.com/prettier) or by sponsoring the projects we depend on, including [typescript-eslint](https://opencollective.com/typescript-eslint), [remark](https://opencollective.com/unified), and [Babel](https://opencollective.com/babel).
```

> If you enjoy Prettier and would like to support our work, head to our [OpenCollective](https://opencollective.com/prettier). Please consider also supporting the projects Prettier depends on, such as [typescript-eslint](https://opencollective.com/typescript-eslint), [remark](https://opencollective.com/unified), and [Babel](https://opencollective.com/babel).

Reviewer: ```suggestion
If you enjoy Prettier and would like to support our work, consider supporting us directly via [our OpenCollective](https://opencollective.com/prettier) or by supporting the projects we depend on, including [typescript-eslint](https://opencollective.com/typescript-eslint), [remark](https://opencollective.com/unified), and [Babel](https://opencollective.com/babel).
```
